selfies/utils/encoding_utils.py

```diff
@@ -48,7 +48,19 @@ def selfies_to_encoding(
 
     # integer encode
     char_list = split_selfies(selfies)
-    integer_encoded = [vocab_stoi[char] for char in char_list]
+
+    try:
+        integer_encoded = [vocab_stoi[char] for char in char_list]
+    except KeyError as e:
+        if e.args[0] == ".":
```

Comment:
Thanks for this PR! Sorry, I would prefer to merge the first commit since it reads more clearly! 

```python
char_list = list(split_selfies(selfies))

# Check if SELFIES string contains unconnected molecules
if "." in list(char_list) and not "." in vocab_stoi:
    raise ValueError(
        "The SELFIES string contains two unconnected molecules "
        "(given by the '.' character), but vocab_stoi does not "
        "contain the '.' key. Please add it or separate the molecules."
    )

integer_encoded = [vocab_stoi[char] for char in char_list]
```    

I agree that the first cast to `char_list` is necessary, but I think there is a redundant one in the if-statement. Alternatively, maybe a one-pass approach as follows could work:

```python
integer_encoded = []

for char in split_selfies(selfies):
    if (char == ".") and ("." not in vocab_stoi):
        raise ValueError("...")
    integer_encoded.append(vocab_stoi[char])
```

Reply:
That sounds better! I've added the suggested one-pass approach in a new commit to make it easier to merge. It passes all the tests, so feel free to merge it or let me know if I can help with anything else!